src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java

```diff
@@ -887,4 +890,37 @@ default boolean shouldDisplayFluidOverlay(BlockState state, IBlockDisplayReader
     {
         return state.getBlock() instanceof BreakableBlock || state.getBlock() instanceof LeavesBlock;
     }
+    
+    /**
+     * Returns the state that this block should transform into when right clicked by a tool.
+     * Used to determine if an axe can strip, a shovel can path, or a hoe can till.
+     * Return null if vanilla behavior should be disabled.
+     *
+     * @param state The current state
+     * @param world The world
+     * @param pos The block position in world
+     * @param player The player clicking the block
+     * @param stack The stack being used by the player
+     * @return The resulting state after the action has been performed
+     */
+    @Nullable
+    default BlockState getToolModifiedState(BlockState state, World world, BlockPos pos, PlayerEntity player, ItemStack stack)
+    {
+    	Set<ToolType> tool = stack.getToolTypes();
+    	if(tool.contains(ToolType.AXE)) {
+    		Block block = AxeItem.BLOCK_STRIPPING_MAP.get(state.getBlock());
+    		if(block == null && tool.size() == 1) return null;
+    		if(block != null) return block.getDefaultState().with(RotatedPillarBlock.AXIS, state.get(RotatedPillarBlock.AXIS));
+    	}
+    	if(tool.contains(ToolType.HOE)) {
+    		BlockState modifiedState = HoeItem.HOE_LOOKUP.get(state.getBlock());
+    		if(modifiedState == null && tool.size() == 1) return null;
+    		if(modifiedState != null) return modifiedState;
+    	}
+    	if(tool.contains(ToolType.SHOVEL)) {
+    		BlockState modifiedState = ShovelItem.SHOVEL_LOOKUP.get(state.getBlock());
+    		return modifiedState;
+    	}
+    	return null;
```

Comment:
Return the original state if you can't find a modified one, not null.

Reply:
The reason I return null is that AxeItem, HoeItem, and ShovelItem all check if the block isn't in the list by comparing it to null.

Reply:
Hmm, that's fair enough.